### PR TITLE
CORE: split mc into mc and ec

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Move UCC execution engine operations from memory component to a separate component called Execution Component (EC for short)

## Why ?
Memory operations and execution engine operations are loosely coupled, posting tasks to execution engine has nothing to do with memory operations such as memcpy and memalloc. Multiple EC can be implemented to work with execution engines of different types without touching existing MC implementation

## How ?
Move code from core MC and MC CUDA to EC and EC CUDA

### CODE REVIEW

File: .github/workflows/codestyle.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
               return 1
             fi
           fi
-          if ! echo $msg | grep -qP '^Merge |^((CODESTYLE|REVIEW|CORE|UTIL|TEST|API|DOCS|TOOLS|BUILD|MC|SCHEDULE|TOPO)|(CL/|TL/|MC/|UCP|UCG|NCCL|SHARP|BASIC|HIER|CUDA|CPU|EE))+: \w'
+          if ! echo $msg | grep -qP '^Merge |^((CODESTYLE|REVIEW|CORE|UTIL|TEST|API|DOCS|TOOLS|BUILD|MC|EC|SCHEDULE|TOPO)|(CL/|TL/|MC/|UCP|UCG|NCCL|SHARP|BASIC|HIER|CUDA|CPU|EE))+: \w'
```
Comment (vspetrov) [inline .github/workflows/codestyle.yaml:39]: can we wrap this line?

File: src/Makefile.am [hunk 2]
```diff
@@ -85,6 +90,7 @@ libucc_la_SOURCES =                   \
 	core/ucc_version.c                \
 	core/ucc_context.c                \
 	core/ucc_mc.c                     \
+	core/ucc_ec.c                     \
```
Comment (vspetrov) [inline src/Makefile.am:93]: generic question (i didn't honestly understand that from day one): why ucc_mc.c(h) and ucc_ec.c(h) are in "core" and not under src/components/mc and src/components/ec respectively?

File: src/components/ec/cuda/ec_cuda.c [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+    }
+    return UCC_OK;
+}
+
+ucc_status_t ucc_ec_cuda_task_post(void *ee_stream, void **ee_req)
+{
+    ucc_ec_cuda_stream_request_t *req;
+    ucc_ec_cuda_event_t *cuda_event;
+    ucc_status_t status;
+    ucc_ec_cuda_config_t *cfg = EC_CUDA_CONFIG;
```
Comment (vspetrov) [inline src/components/ec/cuda/ec_cuda.c:225]: init var should go first

File: src/components/ec/cuda/kernel/Makefile.am [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+#
+
+NVCCFLAGS = ${AM_CPPFLAGS} ${UCS_CPPFLAGS} ${NVCC_CFLAGS} -I${UCC_TOP_BUILDDIR} -I${UCC_TOP_SRCDIR}/src -I${UCC_TOP_BUILDDIR}/src --compiler-options -fno-rtti,-fno-exceptions
```
Comment (vspetrov) [inline src/components/ec/cuda/kernel/Makefile.am:5]: plz wrap the line

**Comment by Sergei-Lebedev** [comment]
@vspetrov fixed

File: src/Makefile.am (hunk 2 above)
Comment (Sergei-Lebedev) [reply src/Makefile.am]: agree

**Review by vspetrov — APPROVED** [review]

**Review by manjugv — APPROVED** [review]

**Comment by Sergei-Lebedev** [comment]
bot:retest

**Comment by Sergei-Lebedev** [comment]
bot:retest

**Pull request merged**